### PR TITLE
Arm SVE Vectorization

### PR DESCRIPTION
**Pull request by sscholbe**
This is the pull request for implementing the vectorization transform for the Arm SVE codegen.

### CODE REVIEW

File: dace/transformation/dataflow/sve/vectorization.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+    @classmethod
+    def expressions(cls):
+        return [sdutil.node_path_graph(cls.map_entry)]
+
+    @classmethod
+    def can_be_applied(cls, graph, candidate, expr_index, sdfg, strict=False):
+        entry_node = graph.node(candidate[cls.map_entry])
+        exit_node = graph.exit_node(entry_node)
+        nodes_between = graph.all_nodes_between(entry_node, exit_node)
```
Comment (tbennun) [inline dace/transformation/dataflow/sve/vectorization.py:34]: use `graph.scope_subgraph`

File: dace/transformation/dataflow/sve/vector_inference.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+from samples.fpga.jacobi_fpga_systolic import T
```
Comment (sscholbe) [inline dace/transformation/dataflow/sve/vector_inference.py:1]: Classic VSCode

File: dace/transformation/dataflow/sve/vectorization.py [hunk 3]
```diff
@@ -0,0 +1,212 @@
+# Copyright 2019-2021 ETH Zurich and the DaCe authors. All rights reserved.
+"""
+    SVE Vectorization: This module offers all functionality to vectorize an SDFG for the Arm SVE codegen.
+"""
+from tests.buffer_tiling_test import I
```
Comment (sscholbe) [inline dace/transformation/dataflow/sve/vectorization.py:5]: +1

File: dace/codegen/targets/sve/codegen.py [hunk 4]
```diff
@@ -77,6 +80,44 @@ def get_generated_codeobjects(self):
...
+                                                          edge.data,
+                                                          not nc,
+                                                          None,
+                                                          '@',
+                                                          dtype=desc.dtype)
+            callsite_stream.write(
+                wcr[:wcr.find('@')] +
+                util.REDUCTION_TYPE_TO_SVE[reduction_type] +
+                f'(svptrue_{util.TYPE_TO_SVE_SUFFIX[desc.dtype]}(), ' +
+                src_node.label + wcr[wcr.find('@') + 1:] + ');')
```
Comment (tbennun) [inline dace/codegen/targets/sve/codegen.py:111]: strange choice of implementation. please document why you do that!

File: dace/codegen/targets/sve/unparse.py [hunk 5]
```diff
@@ -331,9 +331,14 @@ def vector_reduction_expr(self, edge, dtype, rhs):
             # WCR on vectors works in two steps:
             # 1. Reduce the SVE register using SVE instructions into a scalar
             # 2. WCR the scalar to memory using DaCe functionality
+            dst_node = self.dfg.memlet_path(edge)[-1].dst
+            if isinstance(dst_node, nodes.AccessNode) and dst_node.desc(self.sdfg).storage == dtypes.StorageType.SVE_Register:
+                print('wixx')
+                return
```
Comment (tbennun) [inline dace/codegen/targets/sve/unparse.py:337]: ?

**Review by tbennun — CHANGES_REQUESTED** [review]
Minor changes

**Comment by tbennun** [comment]
see #883

**Pull request closed without merging**